extensions/Worker.Extensions.EventGrid/src/TypeConverters/EventGridBinaryDataConverter.cs

```diff
@@ -0,0 +1,97 @@
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.EventGrid.TypeConverters
+{
+    /// <summary>
+    /// Converter to bind to BinaryData or BinaryData[] parameter.
+    /// </summary>
+    [SupportedConverterType(typeof(BinaryData))]
+    [SupportedConverterType(typeof(BinaryData[]))]
+    internal class EventGridBinaryDataConverter : IInputConverter
+    {
+        private readonly IOptions<WorkerOptions> _workerOptions;
+
+        public EventGridBinaryDataConverter(IOptions<WorkerOptions> workerOptions)
+        {
+            _workerOptions = workerOptions ?? throw new ArgumentNullException(nameof(workerOptions));
+        }
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (context.TargetType != typeof(BinaryData) && context.TargetType != typeof(BinaryData[]))
+            {
+                return new(ConversionResult.Unhandled());
+            }
+
+            try
+            {
+                var contextSource = context?.Source as string;
+
+                if (contextSource is null)
+                {
+                    return new(ConversionResult.Unhandled());
+                }
+
+                if (context!.TargetType == typeof(BinaryData))
+                {
+                    var binaryData = BinaryData.FromObjectAsJson<string>(contextSource);
+                    return new(ConversionResult.Success(binaryData));
+                }
+                else if (context!.TargetType == typeof(BinaryData[]))
+                {
+                    byte[] byteArray = Encoding.UTF8.GetBytes(contextSource);
+                    MemoryStream stream = new MemoryStream(byteArray);
+
+                    var jsonData = (List<object>?)_workerOptions?.Value?.Serializer?.Deserialize(stream, typeof(List<object>), CancellationToken.None);
+                    List<BinaryData> binaryDataList = new List<BinaryData>();
+
+                    if (jsonData is not null)
+                    {
+                        foreach (var item in jsonData)
+                        {
+                            if (item is not null)
+                            {
+                                var binaryData = new BinaryData(item);
+                                binaryDataList.Add(binaryData);
+                            }
+                        }
+                        return new(ConversionResult.Success(binaryDataList.ToArray()));
```

Comment:
nit on this if-else-staement:

I think this would be cleaner as a switch with a method for  ConvertToBinaryDataArray i.e.

```csharp
var result = context!.TargetType switch
{
    Type t when t == typeof(BinaryData) => BinaryData.FromObjectAsJson<string>(contextSource),
    Type t when t == typeof(BinaryData[]) => ConvertToBinaryDataArray(contextSource),
    _ => null
};

if (result is not null)
{
    return new(ConversionResult.Success(result));
}
```

Reply:
Changed!